src/prefect/events/clients.py

```diff
@@ -346,6 +346,47 @@ async def _emit(self, event: Event) -> None:
                     await asyncio.sleep(1)
 
 
+class AssertingPrefectEventsClient(PrefectEventsClient):
```

Comment:
I am really not happy about this but I don't see another path right now for tests like this: https://github.com/PrefectHQ/prefect/blob/51970458565a2856faf9796b3eb142a27ef82053/tests/test_task_runs.py#L26-L43